core/parentchain/indirect-calls-executor/src/indirect_calls_executor.rs

```diff
@@ -74,41 +86,25 @@ where
 		self.stf_executor.execute_shield_funds(account, *amount, shard)?;
 		Ok(())
 	}
-
-	fn decrypt_unchecked_extrinsic(
-		&self,
-		xt: UncheckedExtrinsicV4<CallWorkerFn>,
-	) -> Result<(TrustedCallSigned, ShardIdentifier)> {
-		let (call, request) = xt.function;
-		let (shard, cyphertext) = (request.shard, request.cyphertext);
-		debug!("Found CallWorker extrinsic in block: \nCall: {:?} \nRequest: \nshard: {}\ncyphertext: {:?}",
-        	call, bs58::encode(shard.encode()).into_string(), cyphertext);
-
-		debug!("decrypt the call");
-		let shielding_key = self.shielding_key_repo.retrieve_key()?;
-		let request_vec = shielding_key.decrypt(&cyphertext)?;
-
-		Ok(TrustedCallSigned::decode(&mut request_vec.as_slice()).map(|call| (call, shard))?)
-	}
 }
 
-impl<ShieldingKeyRepository, StfExecutor> ExecuteIndirectCalls
-	for IndirectCallsExecutor<ShieldingKeyRepository, StfExecutor>
+impl<ShieldingKeyRepository, StfExecutor, TopPoolAuthor> ExecuteIndirectCalls
+	for IndirectCallsExecutor<ShieldingKeyRepository, StfExecutor, TopPoolAuthor>
 where
 	ShieldingKeyRepository: AccessKey,
 	<ShieldingKeyRepository as AccessKey>::KeyType:
 		ShieldingCryptoDecrypt<Error = itp_sgx_crypto::Error>,
-	StfExecutor: StfExecuteTrustedCall + StfExecuteShieldFunds,
+	StfExecutor: StfExecuteShieldFunds,
+	TopPoolAuthor: AuthorApi<H256, H256> + Send + Sync + 'static,
 {
 	fn execute_indirect_calls_in_extrinsics<ParentchainBlock>(
```

Comment:
```suggestion
	fn submit_indirect_calls_to_top_pool<ParentchainBlock>(
```

Reply:
We're currently doing a mix of executing (shield funds) and submitting to top pool. Is it okay for you if we do this re-naming with #755 ?

Reply:
That's fine :)